zeebe/bpmn-model/src/test/java/io/camunda/zeebe/model/bpmn/validation/ZeebeTaskListenersValidationTest.java

```diff
@@ -91,8 +95,36 @@ void testEventTypeNotSupported(final ZeebeTaskListenerEventType unsupportedEvent
             ZeebeTaskListener.class,
             String.format(
                 "Task listener event type '%s' is not supported. "
-                    + "Currently, only 'assigning', 'completing' event types "
-                    + "and 'assignment', 'complete' deprecated event types are supported.",
+                    + "Currently, only 'assigning', 'updating', 'completing' event types "
+                    + "and 'assignment', 'update', 'complete' deprecated event types are supported.",
                 unsupportedEventType)));
   }
+
+  @DisplayName("task listener with supported `eventType` property")
+  @ParameterizedTest(name = "supported event type: ''{0}''")
+  @EnumSource(
+      value = ZeebeTaskListenerEventType.class,
+      // supported event types
+      names = {
+        "assigning", "updating", "completing",
+        // deprecated event types
+        "assignment", "update", "complete"
+      })
+  void testEventTypeSupported(final ZeebeTaskListenerEventType supportedEventType) {
+    // given
+    final BpmnModelInstance process =
+        Bpmn.createExecutableProcess("process")
+            .startEvent()
+            .userTask(
+                "user_task",
+                ut ->
+                    ut.zeebeUserTask()
+                        .zeebeTaskListener(
+                            l -> l.eventType(supportedEventType).type("supported_listener")))
```

Comment:
🔧 I needed to look up, what this "type" attribute is good for as the naming is very abstract. As I understood, this is the key/name of the job that will be scheduled. Thus maybe we could call it `some_job`?